### PR TITLE
test

### PR DESCRIPTION
**Pull request by chia7712**
let's see what happens :)

### CODE REVIEW

**Comment by mumrah** [comment]
Hm, looks like it didn't trigger. Let me try to find out why.

**Comment by mumrah** [comment]
@chia7712 I think I know why it didn't trigger. The workflow is currently configured to trigger on "push" and "pull_request" events.

The "push" event happens any time you push code to a remote branch. For your `gh-parallel` branch, when you pushed to your personal fork, an action did trigger https://github.com/chia7712/kafka/actions/runs/10464210138. However, we don't see it on the apache/kafka PR. 

There is another type of event "pull_request_target" which fires when a public fork creates a PR _targeting_ apache/kafka. This is probably want we want if we want to keep the same build policy that we have with Jenkins (where public forks can trigger builds). I'll look into this today.

In the mean time, you can push changes to apache/kafka "gh-*" branches and open PRs from there.

**Comment by chia7712** [comment]
@mumrah thanks for sharing. it seems we don't get much benefits from removing `PmaxParallelForks`

*before*: 2 h
https://github.com/apache/kafka/actions/runs/10464006813 

*after*: 1h 59m
https://github.com/chia7712/kafka/actions/runs/10464210138

will test it again

**Comment by chia7712** [comment]
> There is another type of event "pull_request_target" which fires when a public fork creates a PR targeting apache/kafka. This is probably want we want if we want to keep the same build policy that we have with Jenkins (where public forks can trigger builds). I'll look into this today.

I guess the root cause is that we require the branch "gh-". Hence, only the commit targeted into "gh-" can be triggered. I have changed the branch limit from "gh-" to "trunk", and it works now.

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -16,12 +16,12 @@
...
 
 on:
-  push:
-    branches: 'gh-*'
-
   pull_request:
-    types: [opened, synchronize, reopened]
-    branches: 'gh-*'
+    branches:
+      - trunk
```
Comment (chia7712) [inline .github/workflows/pr.yml:21]: This means we run CI only for the commit targeted into "trunk"

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
+  push:
+    branches:
+      - trunk
```
Comment (chia7712) [inline .github/workflows/pr.yml:24]: this means we run CI for each commit pushed into trunk. (a kind of post-commit)

**Comment by mumrah** [comment]
Alright, I'm starting to understand things a bit better. This PR triggered a "pull_request" "synchronize" action 
image

Even though the workflow includes the Develocity secret injection, the build scan publish failed. I believe this is because chia7712 does not define the GE_ACCESS_TOKEN secret.

> Due to the dangers inherent to automatic processing of PRs, GitHub’s standard pull_request workflow trigger by default prevents write permissions and secrets access to the target repository.

If we ran this as a `pull_request_target` workflow, that secret _would_ be available. 

> Workflows triggered via pull_request_target have write permission to the target repository. They also have access to target repository secrets.

However, the `pull_request_target` workflow would checkout the code from apache/kafka trunk (or whatever the target/base branch was) rather than `chia7712:gh-parallel`. It is possible to override the checkout action to actually checkout the PR code, but this is unsafe (many examples of this online).

Since we probably don't care about build scans from PRs (after all, many will have failures during development), we can probably just use the regular `pull_request` event for builds. I'll update #16936  

Quotes from https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/

**Comment by chia7712** [comment]
close as #16936 is almost completed

**Pull request closed without merging**